File: core/models/recommendation.py
from pydantic import BaseModel
from typing import List

class Recommendation(BaseModel):
    video_id: str
    video_title: str
    video_date: str
    video_duration: str
    playlist_id: str
    playlist_name: str
    source: str
    reason: str
    level: str

class RecommendationList(BaseModel):
    recommendations: List[Recommendation]

File: frontend/types/Recommendation.ts
import { z } from "zod"

export const Recommendation = z.object({
    video_id: z.string(),
    video_title: z.string(),
    video_date: z.string(),
    video_duration: z.string(),
    playlist_id: z.string(),
    playlist_name: z.string(),
    source: z.string(),
    reason: z.string(),
    level: z.string(),
})